### ISSUE TITLE
Map Blank

### ISSUE THREAD

**Issue by jessica-azgs** [StraboSpot/strabo-mobile#154]
When selecting a spot from the map, opening the spot, clicking on the georeference tab then clicking on `Set from Map` the map that opens is blank (white screen except for header bar). This doesn't happen if you open a spot from the spots list then do `Set from Map`.

Getting the same issue when trying to go straight to the map from the login page to the map or after saving a spot.

I think there is some problem with the `MapView` code. If we set a breakpoint in `MapView.getMapView()` before the new map view is called the map is fine.

**Comment by ginxwar** [comment]
I spent some time on this and the problem isn't at ```MapView```.  If you debug a bit further, the map actually sticks around long enough until it reaches a point in ionic.bundle.js (L: 13982), where ionic decides it is safe to garbage collect and toss the entire thing away.  Very bad.

The issue has something to do with ionic and the construction/destruction of the nav-view when it moves from state to state.

I've got a temporary fix for this but this fix has exposed some bad code on my drawing tools.  The drawing interaction isn't properly removed from the map after it reaches ```draw.on('drawend', ...)```

I hope to finish this by tomorrow.

**Comment by ginxwar** [comment]
Found the problem.  The issue is during ionic's view transition of the state ```$ionicView.unloaded```.  When this is triggered, the view's controller has been destroyed and its element has been removed from the DOM.

When the app loads, you are automatically sent to the map.  Then when you select a spot from the spots list, you click on the navigator and go to the spots list, invoking the unload event of the map.  This is why when the georeference "Set from map" works, because the map has been completely unloaded and ionic  ends up rebuilding it.

However, upon app load into the map, and you click on the spot to go to the georeference tab, the unload event is **never** triggered.  So when you finally get back to the map, ionic is confused and thinks a transition has just occurred and runs the unload event AFTER it loads the map, which destroys everything.

**Comment by ginxwar** [comment]
I've narrowed the issue.  It looks like the problem may be related to the ol3 popup, specifically in https://github.com/StraboSpot/strabo-mobile/blob/master/www/js/controllers/map.js#L966.  The problem is that this link inside the popup isn't bound to the angular scope, and by extension, would not be bound to anything that ionic could track.

You can prove this by changing the link to something like this:
```
<a href="javascript:void(0);" onclick="$scope.runThisFunction();">
```
Clicking on this would throw an error, saying that $scope is undefined.

I think there are two solutions/workarounds:

* figure out how to get the link talking to angular
* figure out how to *manually* trigger a ```$ionicView.unloaded``` view state on the target page.  I've started a forum topic for help here http://forum.ionicframework.com/t/manually-trigger-ionicview-unloaded/26863

Worst case, maybe the html content within the popup is designed to be informational only and we have to rethink our strategy with map popups.

**Comment by jessica-azgs** [comment]
@ginxwar Remember that this also happens if you try to go straight to the map from the login page or from saving a spot, so it's not just related to the popups. And going from the popup to spot to set-from-map actually still works in strabo.usgin.org. That's an old version - from June 17th (see the About page). But going straight to the map from the login never worked.

**Comment by ginxwar** [comment]
The strabo.usgin.org version was still using the old spot controller which had all the states like 'app.details' (/spots/:spotId/details) and 'app.notes' (/spots/:spotId/notes) which were child states of the main app abstract controller.  So it never had to deal with leaving the current controller scope and into a new controller scope.  Since we refactored into the parent/child spot controllers, this issue has appeared.

Yup, going straight to the map from the login page is showing the same type of error where the ```$ionicView.unloaded``` is prematurely being unloaded.

Still wracking my brain on this...

**Comment by jessica-azgs** [comment]
Finally fixed with commit bf38dde
